SalesforceSDK/Core/Net/HttpCall.cs

```diff
@@ -314,20 +323,24 @@ public static HttpCall CreatePost(string url, string requestBody)
                 }
             }
             HttpResponseMessage message;
+
             try
             {
                 message = await _webClient.SendAsync(req);
-                HandleMessageResponse(message);
             }
-            catch (Exception ex)
+            catch (HttpRequestException ex)
             {
-                _httpCallErrorException = ex;
+                _httpCallErrorException = new DeviceOfflineException("Request failed to send, most likely because we were offline", ex);
```

Comment:
We do make an assumption here that if the actual send fails that the request couldn't be made in the first place (either due to a timeout, disconnect, call cancelled, etc). Since we don't cancel httprequests in this class we can rule out the cancel but there are no way otherwise for us to distinguish WHY the request failed to go out (big limitation of HttpClient IMO)